packages/themeables/core/src/index.ts

```diff
@@ -3,19 +3,37 @@ import type { FC } from 'react'
 import { component, startWithType } from 'refun'
 
 export type TThemeables<ThemeType, ComponentMappings> = { [key in keyof ComponentMappings]: (props: ComponentMappings[key]) => ThemeType }
+export type TOverrideables<ThemeType, ComponentMappings> = { [key in keyof ComponentMappings]?: (props: ComponentMappings[key]) => Partial<ThemeType> }
+export type TOverrideContext<ThemeType, ComponentMappings> = React.Context<TOverrideables<ThemeType, ComponentMappings>>
 
-export const setupTheme = <ThemeType, ComponentMappings>(defaultTheme: TThemeables<ThemeType, ComponentMappings>) => {
+export const setupTheme = <ThemeType, ComponentMappings>(
+  defaultTheme: TThemeables<ThemeType, ComponentMappings>,
+  overrideTheme = {} as TOverrideContext<ThemeType, ComponentMappings>
+) => {
   const ThemePiece = createContext(defaultTheme)
 
   type K = keyof ComponentMappings
 
   const createThemeable = <P extends ThemeType>(name: K, Target: FC<any>) => {
     const Themeable = component(
       startWithType<Partial<P> & ComponentMappings[K]>(),
-      (props) => ({
-        ...useContext(ThemePiece)[name](props),
-        ...props,
-      })
+      (props) => {
+        const themeProps = useContext(ThemePiece)[name](props)
+        const overrideCtx = useContext(overrideTheme)
```

Comment:
can we use the full name? "overrideContext" ? 
for me its easier to read, wdyt?